workspaces/ui-v2/src/spectacle-implementations/cloud-viewer.tsx

```diff
@@ -0,0 +1,163 @@
+import { CapturesServiceStore } from '../optic-components/hooks/useCapturesHook';
+import { IOpticContext } from '@useoptic/spectacle';
+import { ChangelogPages } from '../optic-components/pages/changelog/ChangelogPages';
+
+export default function CloudViewer() {
+  const match = useRouteMatch();
+  const params = useParams<{ specId: string }>();
+  const { specId } = params;
+  const task: InMemorySpectacleDependenciesLoader = async () => {
+    const loadExample = async () => {
+      const apiBase = process.env.REACT_APP_API_URL;
+      const response = await fetch(`${apiBase}/api/specs/${specId}`, {
+        headers: { accept: 'application/json' },
+      });
+      if (!response.ok) {
+        throw new Error(`could not find spec ${specId}`);
+      }
+      const responseJson = await response.json();
+      let signedUrl = responseJson.read_url;
+
+      if(!signedUrl){
+        throw new Error(`No read url found: ${JSON.stringify(responseJson)}`);
+      }
+
+      let contentReq = await fetch(signedUrl);
+      if(!contentReq.ok){
+        throw new Error(`Unable to fetch spec ${specId}`);
+      }
+
+      let spec = await contentReq.json();
+      return spec;
+    };
+    const [example, opticEngine] = await Promise.all([
+      loadExample(),
+      import('@useoptic/diff-engine-wasm/engine/browser'),
+    ]);
+    return {
+      events: example.events,
+      samples: example.session.samples,
+      opticEngine,
+    };
+  };
+  const { loading, error, data } = useInMemorySpectacle(task);
+  if (loading) {
+    return <Loading />;
+  }
+  if (error) {
+    return <div>error :(</div>;
+  }
+  if (!data) {
+    return <div>something went wrong</div>;
+  }
+
+  return (
+    <SpectacleStore spectacle={data}>
+      <CapturesServiceStore capturesService={data.opticContext.capturesService}>
+        <InMemoryInteractionLoaderStore samples={data.samples}>
+          <BaseUrlProvider value={{ url: match.url }}>
+            <Switch>
+              <>
+                <DiffReviewEnvironments />
+                <DocumentationPages />
+                <ChangelogPages />
+              </>
+            </Switch>
+          </BaseUrlProvider>
+        </InMemoryInteractionLoaderStore>
+      </CapturesServiceStore>
+    </SpectacleStore>
+  );
+}
+
+export interface InMemorySpectacleDependencies {
+  events: any[];
+  opticEngine: any;
+  samples: any[];
+}
+
+export type InMemorySpectacleDependenciesLoader = () => Promise<InMemorySpectacleDependencies>;
+
+class InMemorySpectacle implements IForkableSpectacle, InMemoryBaseSpectacle {
+  private spectaclePromise: Promise<any>;
+
+  constructor(
+    public readonly opticContext: IOpticContext,
+    public samples: any[],
+  ) {
+    this.spectaclePromise = makeSpectacle(opticContext);
+  }
+
+  async fork(): Promise<IBaseSpectacle> {
+    const opticContext = await InMemoryOpticContextBuilder.fromEventsAndInteractions(
```

Comment:
there's another builder that doesn't use interactions, "fromEvents", that you may wish to use here, or make a new similar one

Reply:
Used it! :)